python/tests/test_lowlevel.py

```diff
@@ -387,28 +387,30 @@ def test_fromdict_bad_args(self):
 
 class TestIbd:
     def test_uninitialised(self):
-        result = _tskit.IbdResult.__new__(_tskit.IbdResult)
+        result = _tskit.IbdSegments.__new__(_tskit.IbdSegments)
         with pytest.raises(SystemError):
             result.get(0, 1)
         with pytest.raises(SystemError):
             result.print_state()
         with pytest.raises(SystemError):
```

Comment:
Is there a reason why these new attributes are uninitialised (as opposed to 0)?

Reply:
This is reflecting a tricky detail about how we're managing memory. The low-level IbdSegments class is a thin wrapper around the ``tsk_ibd_segments_t`` class in C. We only ever create a new IbdSegments class as a result of calling ``tables.find_ibd()``, which ensures that the memory management is handled correctly. What this test is doing is ensuring that if someone deliberately sidesteps all this, and tries to create an IbdSegments instance manually they don't get a segfault (which they would, if we weren't carefully checking the state of things).

Basically this will never happen, it's just low-level stuff to make sure the memory management is all water tight.